File: python/2014/interstellar-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Interstellar to the database
movies.insert(
    title="Interstellar", 
    year=2014, 
    plot="A team of explorers travel through a wormhole in space in an attempt to ensure humanity's survival.", 
    rating=8.6
)

# Confirm that the movie was added
movie = movies.select(
    title="Interstellar", 
    year=2014
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    

File: csharp/2014/interstellar-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Interstellar" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Interstellar",
                year: 2014,
                plot: "A team of explorers travel through a wormhole in space in an attempt to ensure humanity's survival.",
                rating: 8.6
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Interstellar",
                year: 2014
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}
